jorldy/process.py

```diff
@@ -53,44 +52,47 @@ def manage_process(
     path_queue.put(log_manager.path)
     config_manager.dump(log_manager.path)
 
-    step, print_stamp, eval_thread = 0, 0, None
+    heap = {"step": 0, "run_step": run_step, "wait_thread": False, "wait_process": True}
+    step = 0
+    gath_thread = Thread(
+        target=gather_thread,
+        args=(result_queue, metric_manager, heap, "append"),
+    )
+    gath_thread.start()
```

Comment:
The manager updates the weights when eval starts, so there is no need to share data on the heap.
In order to reuse gather_thread, it seems better to use the existing method with a little modification.